src/Main.hs

```diff
@@ -13,19 +13,42 @@ import qualified Data.Set as Set
 
 default (T.Text)
 
+data Config
+  = Config
+  { _patches   :: FP.FilePath -- ^ path to the @patches@ folder. Should contain @<package-id>-<package-version>.patch@ files.
+  , _keys      :: FP.FilePath -- ^ path to the keys, as generated with the @hackage-repo-tool@.
+  , _template  :: FP.FilePath -- ^ template repo. This will be copied into the temporary repo, and can contain additional files as needed.
+  , _cabal_cfg :: FP.FilePath -- ^ the cabal config file to use. This will be used to download a copy of the base hackage repository from on which this overlay will be based.
+  , _remote_repo_cache :: FP.FilePath -- ^ path to the package cache ( should match the one in the cabal.cfg )
+  , _remote_repo_name  :: String   -- ^ name of the remote repo
+  --
+  , _tar_cmd :: FP.FilePath -- ^ name of the @tar@ command.
+  , _rsync_target :: Text   -- ^ e.g. user@host:/path/to/repo/
+  }
+  deriving (Show)
+
 main :: IO ()
 main =  do
       hSetBuffering stdout LineBuffering
-      shelly $ verbosely $ do
+      shelly $ verbosely $ mkOverlay (Config
+                                      "patches"
+                                      ".keys"
+                                      ".tmpl"
+                                      "cabal.cfg"
+                                      "packages"
+                                      "hackage.haskell.org"
```

Comment:
Please make these configurable via CLI (e.g. optparse-applciative), as they're obviously specific to the installation (i.e. gtar/urls). If you want, you can make the foldernames configurable as well, but it's not that essential (as one can symlink as workaround).

Reply:
I think this is resolved now?